probe-rs/src/architecture/arm/communication_interface.rs

```diff
@@ -101,6 +101,8 @@ pub struct Initialized {
     dps: HashMap<DpAddress, DpState>,
     use_overrun_detect: bool,
     sequence: Arc<dyn ArmDebugSequence>,
+    // This map maps from AP number to TAR value.
+    current_tar: HashMap<u8, u32>,
```

Comment:
This should probably be in a different location, the CSW e.g. is saved here: [cached_csw_value](https://github.com/probe-rs/probe-rs/blob/1f984f1ae4af352e53aceb6779b130ccfd81a363/probe-rs/src/architecture/arm/memory/adi_v5_memory_interface.rs#L59).

Not sure what the best location for this would be, but I don't think it's here :).

Reply:
Good point! I'll have a look!

Reply:
This is trickier than expected because the logging and the actual read of the value happens a level deeper. Generally this huge nesting is a problem. Maybe I need to tackle that first.